### ISSUE TITLE
Resize options unavailable in blivet-gui

### ISSUE THREAD

**Issue by adatum** [storaged-project/blivet-gui#175]
The Resize option is unavailable (greyed out) even though free space is available for increasing lvmlm size. This remains true even for a non-system (not root or home) lvmlv when it is unmounted. 

Are there conditions under which the Resize option is available/unavailable?

Currently using `blivet-gui` version 2.1.13 on Fedora 32.

**Comment by vojtechtrefny** [comment]
Unfortunately the resizing support in [blivet](https://github.com/storaged-project/blivet) (library blivet-gui is using) is very limited. For example only Ext2/3/4 and NTFS filesystems can be resized and these need to be unmounted and "clean" (fsck runs without errors).

Generally it's hard to tell without logs (you should find them in `/var/log/blivet-gui`) what's wrong in your case.

I'll try to think about a way how to communicate to the user the reason why the device can't be resized, I can see that the grayed out option is not very helpful in this case.

**Comment by adatum** [comment]
The lvm volume I wanted to resize has an ext4 filesystem, and was unmounted. 

Having some feedback why options are available would be helpful.

I checked `/var/log/blivet-gui/blivet.log` and the only errors are:

```
DEBUG:blivet:                       DeviceTree.get_device_by_name returned None
DEBUG:blivet: IGNORED:                                 Caught exception, continuing.
DEBUG:blivet: IGNORED:                                 Problem description: failed to get initiator name from iscsi firmware
DEBUG:blivet: IGNORED:                                 Begin exception details.
DEBUG:blivet: IGNORED:                                     Traceback (most recent call last):
DEBUG:blivet: IGNORED:                                       File "/usr/lib/python3.8/site-packages/blivet/safe_dbus.py", line 139, in call_sync
DEBUG:blivet: IGNORED:                                         ret = connection.call_sync(service, obj_path, iface, method, args,
DEBUG:blivet: IGNORED:                                     gi.repository.GLib.GError: g-io-error-quark: GDBus.Error:org.freedesktop.UDisks2.Error.ISCSI.NoFirmware: No firmware found (36)
DEBUG:blivet: IGNORED:                                     During handling of the above exception, another exception occurred:
DEBUG:blivet: IGNORED:                                     Traceback (most recent call last):
DEBUG:blivet: IGNORED:                                       File "/usr/lib/python3.8/site-packages/blivet/iscsi.py", line 160, in __init__
DEBUG:blivet: IGNORED:                                         initiatorname = self._call_initiator_method("GetFirmwareInitiatorName")[0]
DEBUG:blivet: IGNORED:                                       File "/usr/lib/python3.8/site-packages/blivet/util.py", line 1102, in decorated
DEBUG:blivet: IGNORED:                                         return fn(*args, **kwargs)
DEBUG:blivet: IGNORED:                                       File "/usr/lib/python3.8/site-packages/blivet/iscsi.py", line 194, in _call_initiator_method
DEBUG:blivet: IGNORED:                                         return safe_dbus.call_sync(STORAGED_SERVICE, STORAGED_MANAGER_PATH,
DEBUG:blivet: IGNORED:                                       File "/usr/lib/python3.8/site-packages/blivet/safe_dbus.py", line 145, in call_sync
DEBUG:blivet: IGNORED:                                         raise DBusCallError(msg)
DEBUG:blivet: IGNORED:                                     blivet.safe_dbus.DBusCallError: Failed to call GetFirmwareInitiatorName method on /org/freedesktop/UDisks2/Manager with None arguments: GDBus.Error:org.freedesktop.UDisks2.Error.ISCSI.NoFirmware: No firmware found
DEBUG:blivet: IGNORED:                                 End exception details.
```

The beginning of the log file also had: 

> blivet: Failed to call the update_volume_info method: libstoragemgmt functionality not available

So, I installed package `libstoragemgmt`. The message went away from the log, but I have not noticed any change in `blivet-gui`. Resizing is still unavailable.

**Comment by vojtechtrefny** [comment]
These errors are not related to resizing, please upload the entire log, there might be other reasons for blivet-gui to think the device is not resizable.
The log shouldn't contain any sensitive information but if you don't want to upload it publicly, you can send it to me personally to `vtrefny AT redhat DOT com`

**Comment by adatum** [comment]
@vojtechtrefny email sent.

**Comment by vojtechtrefny** [comment]
Thank you for the logs and sorry it took me so long to reply (I usually don't have much time to work on blivet-gui during the week). I didn't find a reason in the logs for the LVs to not be resizable so I decided to look in the code and I found a bug. #182 contains a small rework of the UI part of the resize support. I'm not 100 % sure it will fix your issue but it should at least provide some more information and reason why blivet-gui thinks your devices are not support. The packit bot create a copr repository with the changes [here](https://github.com/storaged-project/blivet-gui/pull/182#issuecomment-629828208) so if you're feeling lucky you can test.

**Comment by adatum** [comment]
No worries @vojtechtrefny and thank you for taking a look and working on a solution.

I tried version 2.1.14 from #182 and it now does allow resizing the unmounted LV. I get a dialog box with a slider which lets me grow or shrink it. I was not brave enough to actually apply the resizing though. 

I also checked how the resize menu option treats other mounted LVs and partitions. If they are mounted, a dialog informs "Mounted devices cannot be resized." and for swap "Resizing of swap format is currently not supported." Helpful.
